### ISSUE TITLE
vue/component-options-name-casing: An option to allow ":" symbol

### ISSUE THREAD

**Issue by ilyub** [vuejs/eslint-plugin-vue#1809]
**What rule do you want to change?**

vue/component-options-name-casing

**Does this change cause the rule to produce more or fewer warnings?**

fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option: allowAdditionalChars

**Please provide some example code that this change will affect:**

```vue
export default defineComponent({
  name: "some-component",
  components: {
    "m-table:messages": MTable as MTableGeneric<Message>,
    "m-table:comments": MTable as MTableGeneric<Comment>
  },
```

**What does the rule currently do for this code?**

Warns about ":" in "m-table:messages" and "m-table:comments"

**What will the rule do after it's changed?**

Should not warn with allowAdditionalChars: [":"]

**Additional context**

I am using Quasar's "GlobalComponentConstructor<Props, Slots>" type: 
![image](https://user-images.githubusercontent.com/70687432/155783097-1a25f09d-4b6d-4efd-8f2f-a59fcfdf90d2.png)
to define generic components as shown above and would like to use "\<component\>:\<type-parameter\>" convention for naming generic components.
